From ce03901f525031d3295dcd4589875f8c875ff970 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 29 Feb 2008 13:48:17 +0200
Subject: [PATCH] Makefile: Rewrite install target

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Makefile |   46 +++++++++++++++++++++++++++++++++++++++++-----
 1 files changed, 41 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index af517a3..e4002d2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,14 +1,36 @@
 # Files
+
+DESTDIR		=
+prefix		= /usr
+exec_prefix	= $(prefix)
+man_prefix	= $(prefix)/share
+
+PACKAGE		= since
+INSTALL		= /usr/bin/install
+INSTALL_BIN	= $(INSTALL) -m 755
+INSTALL_SUID	= $(INSTALL) -m 4755
+INSTALL_DATA	= $(INSTALL) -m 644
+
+INSTALL_OBJS_BIN   = $(PACKAGE)
+INSTALL_OBJS_MAN1  = *.1
+INSTALL_OBJS_SHARE =
+INSTALL_OBJS_ETC   =
+
+BINDIR		= $(DESTDIR)$(exec_prefix)/bin
+SHAREDIR	= $(DESTDIR)$(prefix)/share/$(PACKAGE)
+LIBDIR		= $(DESTDIR)$(prefix)/lib/$(PACKAGE)
+MANDIR1		= $(DESTDIR)$(man_prefix)/man/man1
+
 SRC       = since.c
 OBJ       = since.o
 TARGET    = since
 MAN       = since.1
-BINDIR    = /usr/local/bin
-MANDIR    = /usr/local/man/man1
+#BINDIR    = /usr/local/bin
+#MANDIR    = /usr/local/man/man1
 
 # Compiler and Flags
 CC        = gcc
-CFLAGS    = -Wall -O2 
+CFLAGS    = -Wall -O2
 RM        = rm -f
 INSTALL   = install -c
 LINK      = ln -s
@@ -19,11 +41,11 @@ TAR       = tar
 all : $(TARGET)
 
 # Install
-install : $(TARGET)
+install-old: $(TARGET)
 	$(INSTALL) -m 0755 $(TARGET) $(BINDIR)
 	$(INSTALL) -m 0644 $(MAN) $(MANDIR)
 
-# Build since executable 
+# Build since executable
 $(TARGET) : $(OBJ)
 	$(CC) $(CFLAGS) $(OBJ) -o $(TARGET)
 
@@ -37,3 +59,17 @@ clean :
 
 reallyclean : clean
 	$(RM) $(TARGET)
+
+install-man:
+	# install-man
+	$(INSTALL_BIN) -d $(MANDIR1)
+	$(INSTALL_DATA) $(INSTALL_OBJS_MAN1) $(MANDIR1)
+
+install-bin:
+	# install-bin
+	$(INSTALL_BIN) -d $(BINDIR)
+	$(INSTALL_BIN) -s $(INSTALL_OBJS_BIN) $(BINDIR)
+
+install: all install-bin install-man
+
+.PHONY: clean distclean install install-bin install-man
-- 
1.5.3.8

